File: app/src/main/java/com/hyf/intelligence/iot/widget/dialog/MyDialog.java
package com.hyf.intelligence.iot.widget.dialog;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.hyf.intelligence.iot.R;


public class MyDialog extends Dialog {

    private String title;
    private String content;
    private String leftStr;
    private String rightStr;

    private View.OnClickListener listener;


    public MyDialog(Context context,  String content, View.OnClickListener listener) {
        super(context, R.style.MyDialog);
        this.content = content;
        this.listener = listener;
    }
    public MyDialog(Context context, String title, String content, View.OnClickListener listener) {
        super(context, R.style.MyDialog);
        this.title = title;
        this.content = content;
        this.listener = listener;
    }

    public MyDialog(Context context, String title, String content, String leftText, String rightText, View.OnClickListener listener) {
        super(context, R.style.MyDialog);

        this.title = title;
        this.content = content;
        this.listener = listener;
        this.leftStr = leftText;
        this.rightStr = rightText;


    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_layout);

        TextView tvTitle = findViewById(R.id.title);
        TextView tvContent = findViewById(R.id.content);
        TextView leftText = findViewById(R.id.left_text);
        TextView rightText = findViewById(R.id.right_text);

        if(title == null){
            tvTitle.setVisibility(View.GONE);
        }else{
            tvTitle.setVisibility(View.VISIBLE);
            tvTitle.setText(title);
        }
        tvContent.setText(content);
        if (leftStr != null && !leftStr.equals("")){
            leftText.setText(leftStr);
        }

        if (rightStr != null && !rightStr.equals("")){
            rightText.setText(rightStr);
        }

        leftText.setOnClickListener(listener);
        rightText.setOnClickListener(listener);

    }
}

File: app/src/main/java/com/hyf/iot/widget/dialog/MyDialog.kt
package com.hyf.iot.widget.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.TextView

import com.hyf.iot.R


@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
class MyDialog : Dialog {

    private var title: String? = null
    private var content: String? = null
    private var leftStr: String? = null
    private var rightStr: String? = null

    private var listener: View.OnClickListener? = null


    constructor(context: Context?, content: String, listener: View.OnClickListener) : super(context, R.style.MyDialog) {
        this.content = content
        this.listener = listener
    }

    constructor(context: Context, title: String, content: String, listener: View.OnClickListener) : super(context, R.style.MyDialog) {
        this.title = title
        this.content = content
        this.listener = listener
    }

    constructor(context: Context, title: String, content: String, leftText: String, rightText: String, listener: View.OnClickListener) : super(context, R.style.MyDialog) {
        this.title = title
        this.content = content
        this.listener = listener
        this.leftStr = leftText
        this.rightStr = rightText
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_layout)

        val tvTitle = findViewById<TextView>(R.id.title)
        val tvContent = findViewById<TextView>(R.id.content)
        val leftText = findViewById<TextView>(R.id.left_text)
        val rightText = findViewById<TextView>(R.id.right_text)

        if (title == null) {
            tvTitle.visibility = View.GONE
        } else {
            tvTitle.visibility = View.VISIBLE
            tvTitle.text = title
        }
        tvContent.text = content
        if (leftStr != null && leftStr != "") {
            leftText.text = leftStr
        }

        if (rightStr != null && rightStr != "") {
            rightText.text = rightStr
        }

        leftText.setOnClickListener(listener)
        rightText.setOnClickListener(listener)

    }
}
